solrmonitor/solrmonitor.go

```diff
@@ -155,7 +155,8 @@ func (c *SolrMonitor) doGetCollectionState(name string) (*CollectionState, error
 	if coll == nil {
 		return nil, nil
 	}
-
+	coll.mu.RLock()
```

Comment:
This lock doesn't really protect anything here.

Reply:
That will reflect the latest state of collectionState

Reply:
I suppose it does only return after any write locks are released. Though it's only a pointer getting returned, so the state can change during the use by the consuming function.

Reply:
We update the state inside write lock only (change the pointer to new collectionState). So it should return current state (current pointer to CS)?

Reply:
It's literally protecting the pointer access to `coll.collectionState`.  That pointer is mutable and can point at a different object later.